### ISSUE TITLE
Feature request - Preventing uploading of duplicate files

### ISSUE THREAD

**Issue by tissole** [Nekmo/telegram-upload#114]
I wish that the app could detect duplicate files and skipping them. Maybe storing the names and hashes of files uploaded in a database that keeps track of them. This will still not prevent uploading the files that were already in a channel, uploaded by other means.

So, a solution could be indexing the files from the channel in a database too, for future reference. This option could be turned on/off if the user wishes so.

This feature can be implemented?

**Comment by Nekmo** [comment]
Calculating the hash of the files would imply a significant disk and cpu usage. Research is needed.

**Comment by tissole** [comment]
Obviously, there's no need here for a heavily cryptographic hash algorithm because security is not a concern. A lighter non-cryptographic algorithm can be used and I found one blazing fast [xxHash](https://github.com/Cyan4973/xxHash). It has a python [implementation](https://pypi.org/project/xxhash/) too.

**Comment by Nekmo** [comment]
Of course @tissole , I was thinking about sha1 or sha256.
